### PR TITLE
feat: allow scheme operators

### PR DESCRIPTION
**Pull request by dannydaviesIW**
# Description

-   This PR allows scheme operators to use the service to create the necessary matching json. 

# Testing instructions

-   Pull down this branch and run the site locally. You will need to run the site as two different users (i) an ordinary user with a NOC and (ii) a scheme operator user.
-   To set up your user as a scheme operator, follow these instructions: (i) Delete your user using the cognito manager script in the fdbt-aws repo (ii) Set up your user as a scheme operator using the relevant cognito manager script (iii) Complete registration via email and registration url.
-   Validate that the site works as it used to for an ordinary user, and validate that the site works correctly for a scheme operator.

# Type of change

-   [X] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [X] Able to run pr locally
-   [X] Followed acceptance criteria
-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have made corresponding changes to the documentation if applicable
-   [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/api/salesConfirmation.ts [hunk 1]
```diff
@@ -59,30 +60,35 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
...
-                throw new Error('No period type found to generate user data json.');
-            }
-
-            if (ticketType === 'geoZone') {
-                userDataJson = await getGeoZoneTicketJson(req, res);
-            } else if (ticketType === 'multipleServices') {
-                userDataJson = getMultipleServicesTicketJson(req, res);
+        if (schemeOp) {
+            userDataJson = await getSchemeOperatorTicketJson(req, res);
+        } else if (!schemeOp) {
```
Comment (tejpowar) [inline src/pages/api/salesConfirmation.ts:71]: do we need the if(!schemeOp)?  as would be one or the other right?

File: src/pages/fareType.tsx [hunk 2]
```diff
@@ -125,35 +131,45 @@ const FareTypePage = (props: FareTypeProps): ReactElement => {
...
-    const noc = getAndValidateNoc(ctx);
 
-    if (!operatorCookie || !noc) {
+    if (!operatorCookie || !opIdentifier) {
         throw new Error('Necessary data not found to show faretype page');
     }
     const operatorInfo = JSON.parse(operatorCookie);
-    const uuid = buildUuid(noc);
+    const operatorName = schemeOp ? operatorInfo.operator : operatorInfo.operator.operatorPublicName;
+    opIdentifier = schemeOp ? `${operatorName}` : opIdentifier;
```
Comment (tejpowar) [inline src/pages/fareType.tsx:148]: why is the operator name in string literals?

File: src/pages/fareType.tsx [hunk 3]
```diff
@@ -125,35 +131,45 @@ const FareTypePage = (props: FareTypeProps): ReactElement => {
...
+    const uuid = buildUuid(opIdentifier);
     const cookieValue = JSON.stringify({ ...operatorInfo, uuid });
 
     setCookieOnServerSide(ctx, OPERATOR_COOKIE, cookieValue);
 
-    if (noc !== INTERNAL_NOC) {
+    if (schemeOp || (!schemeOp && opIdentifier !== INTERNAL_NOC)) {
```
Comment (tejpowar) [inline src/pages/fareType.tsx:154]: can this just be || opIdentifier !== INTERNAL_NOC

File: src/pages/api/salesConfirmation.ts (hunk 1 above)
Comment (dannydaviesIW) [reply src/pages/api/salesConfirmation.ts:71]: Good catch, I've changed this so that it is part of the existing else if chain

**Review by tejpowar — APPROVED** [review]

**Pull request merged**